### PR TITLE
[Jira] Add more logging, tests and constants

### PR DESCRIPTION
**Pull request by timgrein**
Some small improvements:
- Add more logging
- Add more tests around `_get_session`
- Catch, if the data source type in `_get_session` is invalid
- Add constants for repeated strings

#### Pre-Review Checklist
- [x] this PR has a meaningful title
~- [ ] this PR links to all relevant github issues that it fixes or partially addresses~
~- [ ] if there is no GH issue, please create it. Each PR should have a link to an issue~
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sources/jira.py [hunk 1]
```diff
@@ -822,7 +836,7 @@ async def _get_projects(self):
                     ):
                         project = await response.json()
                         await self._put_projects(project=project, timestamp=timestamp)
-            await self.queue.put("FINISHED")  # pyright: ignore
+            await self.queue.put(FINISHED)  # pyright: ignore
```
Comment (seanstory) [inline connectors/sources/jira.py:839]: can we get rid of the `# pyright: ignore` while you're here?

Comment (artem-shelkovnikov) [inline connectors/sources/jira.py:839]: Do we actually need pyright: ignore here after this change? What are we even ignoring here? :D

**Review by seanstory — APPROVED** [review]

**Review by artem-shelkovnikov — COMMENTED** [review]
Good stuff!

**Comment by seanstory** [comment]
LOL jinx

**Comment by artem-shelkovnikov** [comment]
What about #pyright: ignore - I see it in several other places changed by this PR, are they needed in all the code you've changed?

**Comment by timgrein** [comment]
> What about #pyright: ignore - I see it in several other places changed by this PR, are they needed in all the code you've changed?

Remove them from my diff

**Review by artem-shelkovnikov — APPROVED** [review]

File: connectors/sources/jira.py [hunk 2]
```diff
@@ -974,9 +990,9 @@ async def get_docs(self, filtering=None):
             )
 
             for rule in advanced_rules:
-                await self.fetchers.put(
-                    partial(self._get_issues, rule.get("query", ""))
-                )
+                query = rule.get("query", "")
+                self._logger.debug(f"Fetching issues using query: {query}")
```
Comment (wangch079) [inline connectors/sources/jira.py:994]: Feel it's more appropriate to put this log inside method `_get_issues`

Comment (timgrein) [reply connectors/sources/jira.py:994]: Sorry, missed your comment, will adapt it 👍

**Pull request merged**